app/forms/webauthn_verification_form.rb

```diff
@@ -55,15 +58,14 @@ def valid_assertion_response?(protocol)
       signature: Base64.decode64(@signature),
     )
     original_origin = "#{protocol}#{self.class.domain_name}"
+    @webauthn_configuration = user.webauthn_configurations.find_by!(credential_id: @credential_id)
+    return false unless @webauthn_configuration
```

Comment:
IIRC, there's no way this method would be able to execute the guard if there's a missing `webauthn_configuration` for the user using a wrong `credential_id`; `@webauthn_configuration` wouldn't never be `nil` (or falsy) because `find_by!` throws an `ActiveRecord::RecordNotFound` exception before get to the next line, so I'd suggest to remove the guard:

```suggestion
    @webauthn_configuration = user.webauthn_configurations.find_by!(credential_id: @credential_id)

```

If this Form needs to be gentle about it, it may need to drop the bang, using `find_by` instead:

```suggestion
    @webauthn_configuration = user.webauthn_configurations.find_by(credential_id: @credential_id)
    return false if @webauthn_configuration.blank?

```

Reply:
nice catch!